torch/_masked/__init__.py

```diff
@@ -686,44 +775,28 @@ def sum(input: Tensor,
         mask: Optional[Tensor] = None) -> Tensor:
     # __doc__ is generated by _apply_docstring_templates decorator
     if dtype is None:
-        dtype = input.dtype
+        # promote integer types to int64 when output dtype is not specified
+        if input.layout == torch.sparse_csr:
+            if input.dtype in {torch.uint8, torch.bool, torch.int8, torch.int16, torch.int32}:
+                # csr.to(dtype=torch.int64) is not implemented, so
```

Comment:
Can you open an issue for this missing feature if it doesn't exist yet? Ideally we just fix this before landing this.

Reply:
Bonus points for simply sending a PR that implements it. It seems like it should be simply about forwarding the call to `to` to the values of the CSR Tensor.

Reply:
This is actually copied from [the comment Pearu added in prod](https://github.com/pytorch/pytorch/blob/master/torch/_masked/__init__.py#L737-L749). Sure I can implement `to` for CSR

Reply:
Just to tie things back together, I'm implementing to.dtype https://github.com/pytorch/pytorch/pull/76400